pallets/proposals/src/impls.rs

```diff
@@ -439,7 +440,33 @@ impl<T: Config> Pallet<T> {
             now,
         ));
 
+        //once the voting is complete check if the milestone is eligible for auto approval
+        //Getting the total threshold required for the milestone to be approved based on the raised funds
+        let funding_threshold: BalanceOf<T> = project.raised_funds * T::PercentRequiredForVoteToPass::get().into()/100u32.into();
+
+        let mut milestone = project.milestones.get_mut(&milestone_key).unwrap().clone();
+        //if the yay votes are both greater than the nay votes and the funding threshold then the milestone is approved
+        if (updated_vote.yay >= updated_vote.nay) && (updated_vote.yay >= funding_threshold) {
+            milestone.is_approved = true;
+             updated_vote = Vote {
+                yay: updated_vote.yay,
+                nay: updated_vote.nay,
+                is_approved: true,
+            };
+            let now = <frame_system::Pallet<T>>::block_number();
+            Self::deposit_event(Event::MilestoneApproved(
+                project.initiator.clone(),
+                project_key,
+                milestone_key,
+                now,
+            ));
+            <MilestoneVotes<T>>::insert((project_key, milestone_key), updated_vote);
+        }
+        project.milestones.insert(milestone_key, milestone.clone());
+        <Projects<T>>::insert(project_key, project);
```

Comment:
Move this to inside the if statement please as there is no need to insert into Projects unless the threshold has been met. 

Also consider using a mutate here instead.